core/corehttp/gateway_handler.go

```diff
@@ -450,7 +437,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 }
 
-func (i *gatewayHandler) servePretty404IfPresent(w http.ResponseWriter, r *http.Request, contentPath ipath.Path) bool {
+// Deprecated: legacy ipfs-404.html files are superseded by _redirects file
+// This is provided only for backward-compatibility, until websites migrate
+// to 404s managed via _redirects file (https://github.com/ipfs/specs/pull/290)
+func (i *gatewayHandler) serveLegacy404IfPresent(w http.ResponseWriter, r *http.Request, contentPath ipath.Path) bool {
```

Comment:
@justincjohnson forgot to push this rename, just pushed, make sure to pull my changes :pray:

Reply:
I've manually added the following locally.  Let me know if I missed anything.

```
diff --git a/core/corehttp/gateway_handler.go b/core/corehttp/gateway_handler.go
index f85c49b1a..e3be79e31 100644
--- a/core/corehttp/gateway_handler.go
+++ b/core/corehttp/gateway_handler.go
@@ -437,7 +437,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
        }
 }
 
-func (i *gatewayHandler) servePretty404IfPresent(w http.ResponseWriter, r *http.Request, contentPath ipath.Path) bool {
+// Deprecated: legacy ipfs-404.html files are superseded by _redirects file
+// This is provided only for backward-compatibility, until websites migrate
+// to 404s managed via _redirects file (https://github.com/ipfs/specs/pull/290)
+func (i *gatewayHandler) serveLegacy404IfPresent(w http.ResponseWriter, r *http.Request, contentPath ipath.Path) bool {
        resolved404Path, ctype, err := i.searchUpTreeFor404(r, contentPath)
        if err != nil {
                return false
diff --git a/core/corehttp/gateway_handler_unixfs__redirects.go b/core/corehttp/gateway_handler_unixfs__redirects.go
index 47c16e807..77d407d91 100644
--- a/core/corehttp/gateway_handler_unixfs__redirects.go
+++ b/core/corehttp/gateway_handler_unixfs__redirects.go
@@ -87,7 +87,7 @@ func (i *gatewayHandler) handlePathResolution(w http.ResponseWriter, r *http.Req
                        // if Accept is text/html, see if ipfs-404.html is present
                        // This logic isn't documented and will likely be removed at some point.
                        // Any 404 logic in _redirects above will have already run by this time, so it's really an extra fall back
-                       if i.servePretty404IfPresent(w, r, contentPath) {
+                       if i.serveLegacy404IfPresent(w, r, contentPath) {
                                logger.Debugw("serve pretty 404 if present")
                                return nil, nil, false
                        }
```